### PR TITLE
some small optimization tweaks

### PR DESCRIPTION
**Pull request by KristofferC**
Some small stuff I found when looking a bit into #336

### CODE REVIEW

File: src/FEValues/common_values.jl [hunk 1]
```diff
@@ -228,13 +230,15 @@ function function_divergence(fe_v::ScalarValues{dim}, q_point::Int, u::AbstractV
...
-function function_divergence(fe_v::VectorValues{dim}, q_point::Int, u::AbstractVector{T}, dof_range::UnitRange = 1:length(u)) where {dim,T}
+function function_divergence(fe_v::VectorValues{dim}, q_point::Int, u::AbstractVector{T}, dof_range = eachindex(u)) where {dim,T}
     n_base_funcs = getn_scalarbasefunctions(fe_v)*dim
     @assert length(dof_range) == n_base_funcs
     @boundscheck checkbounds(u, dof_range)
     diverg = zero(T)
-    @inbounds for (i, j) in enumerate(dof_range)
-        grad = shape_gradient(fe_v, q_point, i)
+    basefunc_count = 1
+    @inbounds for j in dof_range
```
Comment (fredrikekre) [inline src/FEValues/common_values.jl:239]: `enumerate` is used in some other places too, but did this actually make a difference with the other changes?

**Review by fredrikekre — APPROVED** [review]

File: src/FEValues/common_values.jl (hunk 1 above)
Comment (KristofferC) [reply src/FEValues/common_values.jl:239]: Not really, just made the code slightly smaller to look at.

**Comment by kimauth** [comment]
I have few questions just for for the purpose of learning about it. :)

- Do we want use `@propagate_inbounds` instead of `@inline` because these functions are typically used in loops where also using `@inbounds` is a good idea?
- What is better about `eachindex(u)` compared to `1:length(u)`?

**Comment by KristofferC** [comment]
1) yes, we use `@inbounds` for many of the loops where they are called.

2) `eachindex` gives for normal arrays a `OneTo` range where the compiler knows from the type that the first index is 1. When things are inlines and constant propagated it shouldn't matter but it looked like that doesn't happen here (you can look at `@code_warntype` for example).

**Pull request merged**